packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -511,6 +515,19 @@ export class GadgetConnection {
           }
           this.subscriptionClientOptions?.on?.connected?.(socket, payload);
           overrides?.on?.connected?.(socket, payload);
+          activeSocket = socket as globalThis.WebSocket;
+        },
+        ping: (received) => {
+          if (!received) {
+            timedOut = setTimeout(() => {
+              if (activeSocket.readyState === WebSocket.OPEN) {
+                activeSocket.close(4408, "Request Timeout");
+              }
+            }, 2_000);
```

Comment:
I think this might be a bit aggressive for people on planes or trains or what have you with low latency links. Maybe 7s / 3s or something like that? Very finger to the wind

Reply:
Happy to bump up the limits a bit and we can adjust based on future feedback.